### PR TITLE
WLS-313 created a script plugin capable of returning a state object

### PR DESCRIPTION
**Pull request by paul-cioroianu-waylay**
Old script would return everything on state 'Success'. State 'Error' was left unused and it didnt make sense from a plugin point of view since there was no state management.
Now, if the script output is in a plugin format ({observedState:'',rawData:{}} then that value is returned as the plugin output which would enable making use of the 'Error' state in the template designer

### CODE REVIEW

**Review by mfagadar — APPROVED** [review]

**Comment by bo-sec** [comment]
It will be ideal to not use the `Internal team` as reviewers since this team includes persons from multiple teams without any link to this specific repository. Consequently, this generates emails in (almost) everybody's inbox. In such cases, when we don't have code owners for the repositories, we either determine the team which should be the code owners or assign specific persons for the review.

As a note: there are a few repositories which include Internal team as code owners, typically for the ones with general use (e.g. Waylay documentation).

File: script.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "script",
   "version": "2.0.1",
```
Comment (claudiu-muresan-pfa) [inline script.json:3]: When does the script version change? I see we still have it as `2.0.1`

File: plugins/script/index.test.js [hunk 2]
```diff
@@ -22,7 +22,58 @@ async function runPlug (script, options, waylay) {
...
+    .toThrow(new Error('return type of rawData is not an object'))
+})
 
+test('expect output object to have an observable state', async () => {
+  const observedState = 'Success';
+  const rawData = { someData: "sdgerefw" }
+  const options = {
+    requiredProperties: { script: 'function(){return {observedState:\'' + observedState + '\', rawData:' + JSON.stringify(rawData) + '}}' }
+  }
+  console.log(options.requiredProperties.script)
```
Comment (claudiu-muresan-pfa) [inline plugins/script/index.test.js:74]: I guess we can remove the console.log statement

File: plugins/script/index.test.js [hunk 3]
```diff
@@ -22,7 +22,58 @@ async function runPlug (script, options, waylay) {
...
+    .toThrow(new Error('return type of script is not an object'))
+})
+
+test('expect error if observedState is neither "Success" or "Error"', async () => {
+  var options = {
+    requiredProperties: { script: 'function(){return {observedState:"False",rawData:{}}}' }
+  }
+
+  await expect(runPlug(script, options)).rejects
+    .toThrow(new Error('return value of observedState is invalid. Accepted values are [Success|Error]. Current value is ' + 'False'))
```
Comment (claudiu-muresan-pfa) [inline plugins/script/index.test.js:56]: We can drop the concatenation as we know the expected  value 😄 `...Current value is False`

File: plugins/script/index.test.js [hunk 4]
```diff
@@ -22,7 +22,58 @@ async function runPlug (script, options, waylay) {
...
 
-test('', () => {
+test('expect moment.js, uuid and lodash to be available', async () => {
+  const options = {
+    requiredProperties: {
+      script: 'function(){return ' +
+        '{' +
+        'uuid:uuidv4() ? true : false,' +
+        'moment:moment() ? true : false,' +
+        'lodash:_.eq(1,1) ? true : false' +
```
Comment (claudiu-muresan-pfa) [inline plugins/script/index.test.js:32]: Neat. So we now get some great utils when using the script sensor 👍

**Review by claudiu-muresan-pfa — APPROVED** [review]

**Pull request merged**